tools/auto-fuzz/fuzz_driver_generation_jvm.py

```diff
@@ -0,0 +1,292 @@
+                    # Copy other lines from the base fuzzer
+                    content += line
+        return content
+
+
+def _determine_import_statement(classname):
+    """Generate java import statement for a given class name"""
+    primitives = [
+        "boolean", "byte", "char", "short", "int", "long", "float", "double",
+        "void"
+    ]
+
+    if classname and not classname.startswith('java.lang.'):
+        classname = classname.split("$")[0].replace("[]", "")
+        if classname not in primitives:
+            return 'import %s;\n' % classname
+
+    return ''
+
+
+def _handle_import(func_elem):
+    """Loop through the method element and retrieve all necessary import.
+    The key to look for are shown in the following list.
+    1. functionSourceFile
+    2. returnType
+    3. argTypes
+    4. exceptions
+    """
+    import_set = set()
+
+    # functionSourceFile
+    import_set.add(_determine_import_statement(
+        func_elem['functionSourceFile']))
+
+    # returnType
+    import_set.add(_determine_import_statement(func_elem['returnType']))
+
+    # argTypes
+    for argType in func_elem['argTypes']:
+        import_set.add(_determine_import_statement(argType))
+
+    # exceptions
+    for exception in func_elem['JavaMethodInfo']['exceptions']:
+        import_set.add(_determine_import_statement(exception))
+
+    return list(import_set)
+
+
+def _handle_argument(argType, init_dict, possible_target):
+    """Generate data creation statement for given argument type"""
+    if argType == "int" or argType == "java.lang.Integer":
+        return "data.consumeInt(0,100)"
+    elif argType == "int[]" or argType == "java.lang.Integer[]":
+        return "data.consumeInts(100)"
+    elif argType == "boolean" or argType == "java.lang.Boolean":
+        return "data.consumeBoolean()"
+    elif argType == "boolean[]" or argType == "java.lang.Boolean[]":
+        return "data.consumeBooleans(100)"
+    elif argType == "byte" or argType == "java.lang.Byte":
+        return "data.consumeByte()"
+    elif argType == "byte[]" or argType == "java.lang.Byte[]":
+        return "data.consumeBytes(100)"
+    elif argType == "short" or argType == "java.lang.Short":
+        return "data.consumeShort()"
+    elif argType == "short[]" or argType == "java.lang.Short[]":
+        return "data.consumeShorts(100)"
+    elif argType == "long" or argType == "java.lang.Long":
+        return "data.consumeLong()"
+    elif argType == "long[]" or argType == "java.lang.Long[]":
+        return "data.consumeLongs(100)"
+    elif argType == "float" or argType == "java.lang.Float":
+        return "data.consumeFloat()"
+    elif argType == "char" or argType == "java.lang.Character":
+        return "data.consumeCharacter()"
+    elif argType == "java.lang.String":
+        return "data.consumeString(100)"
+    else:
+        return _handle_object_creation(argType, init_dict, possible_target)
+
+
+def _search_conrete_subclass(classname, init_dict):
```

Comment:
nit, should be: `_search_concrete_subclass`

Reply:
changed.